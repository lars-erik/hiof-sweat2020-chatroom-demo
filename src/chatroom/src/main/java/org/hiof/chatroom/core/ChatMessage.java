package org.hiof.chatroom.core;

import java.time.*;
import java.time.format.*;
import java.util.UUID;

public class ChatMessage {
    private UUID id;
    private Instant time;
    private String user;
    private String message;

    public ChatMessage() {
    }

    public ChatMessage(UUID id, Instant time, String user, String message) {
        this.id = id;
        this.time = time;
        this.user = user;
        this.message = message;
    }

    public UUID getId() {
        return id;
    }

    public void setId(UUID id) {
        this.id = id;
    }

    public String getUser() {
        return user;
    }

    public void setUser(String user) {
        this.user = user;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public Instant getTime() {
        return time;
    }

    public void setTime(Instant time) {
        this.time = time;
    }

    @Override
    public String toString() {
        return "ChatMessage{" +
                "time='" + time + '\'' +
                ", user='" + user + '\'' +
                ", message='" + message + '\'' +
                '}';
    }

    DateTimeFormatter formatter = DateTimeFormatter
            .ofPattern("HH:mm:ss")
            .withZone(ZoneId.of("UTC"));

    public String format() {
        String timeString = formatter.toFormat().format(getTime());
        return String.format("%1$s <%2$s>: %3$s", timeString, getUser(), getMessage());
    }
}
