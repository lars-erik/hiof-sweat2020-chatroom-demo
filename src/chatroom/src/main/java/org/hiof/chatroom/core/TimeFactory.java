package org.hiof.chatroom.core;

import java.time.Instant;
import java.util.concurrent.Callable;

public class TimeFactory {
    public static Callable<Instant> nowFactory = () -> Instant.now();
    public static Instant now() throws Exception { return nowFactory.call(); }
    public static void reset() {
        nowFactory = () -> Instant.now();
    }
}
