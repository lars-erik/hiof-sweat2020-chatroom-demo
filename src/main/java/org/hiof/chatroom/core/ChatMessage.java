package org.hiof.chatroom.core;

import java.util.UUID;

public class ChatMessage {
    private UUID id;
    private String user;
    private String message;

    public UUID getId() {
        return id;
    }

    public void setId(UUID id) {
        this.id = id;
    }

    public String getUser() {
        return user;
    }

    public void setUser(String user) {
        this.user = user;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }
}
