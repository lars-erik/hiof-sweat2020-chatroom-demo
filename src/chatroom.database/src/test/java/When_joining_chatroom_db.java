import org.hiof.chatroom.core.When_joining_chatroom;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

public class When_joining_chatroom_db extends When_joining_chatroom {

    @Override
    @BeforeEach
    public void initialize_persistence() throws Exception {
        persistenceSupport = new DbPersistenceSupport();
    }

    @Test
    @Override
    public void last_20_messages_are_shown() throws Exception {
        super.last_20_messages_are_shown();
    }
}
