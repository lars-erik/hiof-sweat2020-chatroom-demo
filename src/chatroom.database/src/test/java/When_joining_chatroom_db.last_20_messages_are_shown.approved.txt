Messages[0] = 23:00:20 <Luke>: Nooo! 20
Messages[1] = 23:00:19 <Luke>: Nooo! 19
Messages[2] = 23:00:18 <Luke>: Nooo! 18
Messages[3] = 23:00:17 <Luke>: Nooo! 17
Messages[4] = 23:00:16 <Luke>: Nooo! 16
Messages[5] = 23:00:15 <Luke>: Nooo! 15
Messages[6] = 23:00:14 <Luke>: Nooo! 14
Messages[7] = 23:00:13 <Luke>: Nooo! 13
Messages[8] = 23:00:12 <Luke>: Nooo! 12
Messages[9] = 23:00:11 <Luke>: Nooo! 11
Messages[10] = 23:00:10 <Luke>: Nooo! 10
Messages[11] = 23:00:09 <Luke>: Nooo! 9
Messages[12] = 23:00:08 <Luke>: Nooo! 8
Messages[13] = 23:00:07 <Luke>: Nooo! 7
Messages[14] = 23:00:06 <Luke>: Nooo! 6
Messages[15] = 23:00:05 <Luke>: Nooo! 5
Messages[16] = 23:00:04 <Luke>: Nooo! 4
Messages[17] = 23:00:03 <Luke>: Nooo! 3
Messages[18] = 23:00:02 <Luke>: Nooo! 2
Messages[19] = 23:00:01 <Luke>: Nooo! 1
